### ISSUE TITLE
sails-auth 1.2.2 breaks sails-permissions' user creation

### ISSUE THREAD

**Issue by khchan** [tjwebb/sails-permissions#58]
On a fresh installation of sails-permissions with sails-auth@1.2.2, the default admin user is created erroneously with the username as 'admin@example.com':

```
debug: sails-permissions: syncing waterline models
debug: sails-permissions: admin user does not exist; creating...
debug: User.afterCreate.setOwner { username: 'admin',
  email: 'admin@example.com',
  createdBy: 1,
  owner: 1,
  model: 4,
  createdAt: Mon May 18 2015 20:27:57 GMT-0400 (EDT),
  updatedAt: Mon May 18 2015 20:27:57 GMT-0400 (EDT),
  id: 1 }
debug: sails-permissions: created admin user: { createdBy: 1,
  owner: 1,
  username: 'admin@example.com',
  email: 'admin@example.com',
  model: 4,
  createdAt: Mon May 18 2015 20:27:57 GMT-0400 (EDT),
  updatedAt: Mon May 18 2015 20:27:58 GMT-0400 (EDT),
  id: 1 }
```

Behaviour was last okay on sails-auth@1.1.0.

**Comment by tjwebb** [comment]
What are you setting the admin username to?

**Comment by khchan** [comment]
It wasn't set before so I just tried again with setting ADMIN_USERNAME, ADMIN_EMAIL and ADMIN_PASSWORD explicitly to the defaults and I still have the same result.

**Comment by tjwebb** [comment]
This should set the username to the value you specify: https://github.com/tjwebb/sails-permissions/blob/master/config/fixtures/user.js#L21

I'll look into it further.

**Comment by ptphats** [comment]
I have same problem. 
At this file: node_modules/sails-auth/api/models/User.js

Remove code:
// beforeValidate: function (user, next) {
  //   if (_.isEmpty(user.username)) {
  //     user.username = user.email;
  //   }
  //   next();
  // },

**Comment by nathanmarks** [comment]
@tjwebb @ptphats That same function is causing me an issue with the `User.afterCreate.setOwner` callback function. It ends up running `beforeValidate` with `{owner: 123}` and strips the username.

**Comment by yshing** [comment]
Same issues here,
Temporary added code in api/models/User.js to override beforeValidate ..

**Comment by tjwebb** [comment]
Resolved in v1.2.3.

see https://github.com/tjwebb/sails-auth/commit/bba666544f4314c97fe161d4331325924ee6b068
